icicle/appUtils/msm/msm.cu

```diff
@@ -479,19 +481,22 @@ void bucket_method_msm(unsigned bitsize, unsigned c, S *scalars, A *points, unsi
   }
   
   //launch the accumulation kernel with maximum threads
-  NUM_THREADS = 1 << 8;
-  NUM_BLOCKS = (h_nof_buckets_to_compute-h_nof_large_buckets + NUM_THREADS - 1) / NUM_THREADS;
-  accumulate_buckets_kernel<<<NUM_BLOCKS, NUM_THREADS, 0, stream>>>(buckets, sorted_bucket_offsets+h_nof_large_buckets, sorted_bucket_sizes+h_nof_large_buckets, sorted_single_bucket_indices+h_nof_large_buckets, point_indices, 
-                                                          d_points, nof_buckets, h_nof_buckets_to_compute-h_nof_large_buckets, c+bm_bitsize, c);                   
+  if (h_nof_buckets_to_compute > h_nof_large_buckets) {
+    NUM_THREADS = 1 << 8;
+    NUM_BLOCKS = (h_nof_buckets_to_compute-h_nof_large_buckets + NUM_THREADS - 1) / NUM_THREADS;
+    accumulate_buckets_kernel<<<NUM_BLOCKS, NUM_THREADS, 0, stream>>>(buckets, sorted_bucket_offsets+h_nof_large_buckets, sorted_bucket_sizes+h_nof_large_buckets, sorted_single_bucket_indices+h_nof_large_buckets, point_indices, 
+                                                            d_points, nof_buckets, h_nof_buckets_to_compute-h_nof_large_buckets, c+bm_bitsize, c);
+  }
+
+  // all the large buckets need to be accumulated before the final summation
   cudaStreamSynchronize(stream2);
-  cudaStreamDestroy(stream2);
```

Comment:
why not destroy the second stream?

Reply:
Tbh I deleted it by accident (along with cudaStreamSyncronize() which gave me a lot of extra debug time lol). I'll put it back